src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -116,8 +119,21 @@ impl<T: 'static> EventLoop<T> {
         let output_manager = OutputManager::new(&env);
 
         // A source of events that we plug into our event loop.
-        let wayland_source = WaylandSource::new(event_queue).quick_insert(event_loop.handle())?;
-        let wayland_source = Rc::new(wayland_source);
+        let wayland_source = WaylandSource::new(event_queue);
+        let wayland_dispatcher = calloop::Dispatcher::new(wayland_source, |_, queue, ddata| {
```

Comment:
What's `ddata`? Does the name have some significance or is this just typoed?

Reply:
it's basically WinitState here, so I'd rename it, just called it that way initially since you have to cast it with older version of library.